File: Server/src/main/content/global/skill/production/crafting/SnakeSkinPlugin.java
package content.global.skill.production.crafting;

import content.global.skill.production.crafting.armour.SnakeSkin;
import content.global.skill.production.crafting.armour.SnakeSkinPulse;
import core.game.dialogue.SkillDialogueHandler;
import core.game.dialogue.SkillDialogueHandler.SkillDialogue;
import core.game.interaction.NodeUsageEvent;
import core.game.interaction.UseWithHandler;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;
import core.plugin.Initializable;
import core.plugin.Plugin;

/**
 * The Snake skin plugin.
 */
@Initializable
public class SnakeSkinPlugin extends UseWithHandler {

    /**
     * Instantiates a new Snake skin plugin.
     */
    public SnakeSkinPlugin() {
        super(1733);
    }

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        addHandler(6289, ITEM_TYPE, this);
        return this;
    }

    @Override
    public boolean handle(final NodeUsageEvent event) {
        final Player player = event.getPlayer();
        new SkillDialogueHandler(player, SkillDialogue.FIVE_OPTION, (Object[]) getSkins()) {

            @Override
            public void create(final int amount, int index) {
                player.getPulseManager().run(new SnakeSkinPulse(player, event.getUsedItem(), amount, SnakeSkin.values()[index]));
            }

            @Override
            public int getAll(int index) {
                return player.getInventory().getAmount(new Item(6289));
            }

            @Override
            public String getName(Item item) {
                return item.getId() == 6328 ? "Boots" : item.getId() == 6330 ? "Vambs" : item.getId() == 6326 ? "Bandana" : item.getId() == 6324 ? "Chaps" : "Body";
            }

        }.open();
        return true;
    }

    /**
     * Get skins item [ ].
     *
     * @return the item [ ]
     */
    public Item[] getSkins() {
        Item[] items = new Item[SnakeSkin.values().length];
        for (int i = 0; i < items.length; i++) {
            items[i] = SnakeSkin.values()[i].getProduct();
        }
        return items;
    }

}

File: Server/src/main/content/global/skill/production/crafting/plugins/SnakeSkinPlugin.kt
package content.global.skill.production.crafting.plugins

import content.global.skill.production.crafting.data.SnakeskinData
import content.global.skill.production.crafting.item.SnakeskinCraftPulse
import core.api.consts.Items
import core.game.dialogue.SkillDialogueHandler
import core.game.interaction.NodeUsageEvent
import core.game.interaction.UseWithHandler
import core.game.node.item.Item
import core.plugin.Initializable
import core.plugin.Plugin

@Initializable
class SnakeSkinPlugin : UseWithHandler(Items.NEEDLE_1733) {

    override fun newInstance(arg: Any?): Plugin<Any> {
        addHandler(Items.SNAKESKIN_6289, ITEM_TYPE, this)
        return this
    }

    override fun handle(event: NodeUsageEvent): Boolean {
        val player = event.player
        object : SkillDialogueHandler(player, SkillDialogue.FIVE_OPTION, (skins as Array<Any?>?)!!) {
            override fun create(amount: Int, index: Int) {
                player.pulseManager.run(
                    SnakeskinCraftPulse(
                        player,
                        event.usedItem,
                        amount,
                        SnakeskinData.values()[index]
                    )
                )
            }

            override fun getAll(index: Int): Int {
                return player.inventory.getAmount(Item(Items.SNAKESKIN_6289))
            }

            public override fun getName(item: Item): String {
                return when (item.id) {
                    Items.SNAKESKIN_BOOTS_6328 -> "Boots"
                    Items.SNAKESKIN_VBRACE_6330 -> "Vambs"
                    Items.SNAKESKIN_BANDANA_6326 -> "Bandana"
                    Items.SNAKESKIN_CHAPS_6324 -> "Chaps"
                    else -> "Body"
                }
            }
        }.open()
        return true
    }

    val skins: Array<Item?>
        get() {
            val items = arrayOfNulls<Item>(SnakeskinData.values().size)
            for (i in items.indices) {
                items[i] = SnakeskinData.values()[i].product
            }
            return items
        }
}
